core_backend/app/routers/manage_language.py

```diff
@@ -0,0 +1,181 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth import get_current_fullaccess_user, get_current_readonly_user
+from ..db.db_models import (
+    LanguageDB,
+    delete_language_from_db,
+    get_default_language_from_db,
+    get_language_from_db,
+    get_list_of_languages_from_db,
+    is_language_name_unique,
+    save_language_to_db,
+    update_language_in_db,
+)
+from ..db.engine import get_async_session
+from ..schemas import AuthenticatedUser, LanguageBase, LanguageRetrieve
+from ..utils import setup_logger
+
+router = APIRouter(prefix="/language")
+logger = setup_logger()
+
+
+@router.post("/create", response_model=LanguageRetrieve)
+async def create_language(
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Create language endpoint. Upsert language to PG database
+    """
+    language.language_name = language.language_name.upper()
+    if not (await is_language_name_unique(language.language_name, asession)):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
```

Comment:
This means that currently it is possible to have no default language. I guess we can prevent that from the frontend, but not sure if we should be allowing it from the backend! What do you think?

Maybe something you can check with @thegeektets ?

If we want to always ensure at least 1 language is set to default, here we can add
```
elif not default_language_exists():
    raise HTTPException(...)
```

Reply:
We can address in another PR when we redeisng APIs

Reply:
I can update that to make sure there is always a default language in the DB. 
About your point, If I remember correctly there is software development [principle](https://www.highlight.io/blog/3-levels-of-data-validation-in-a-full-stack-application-with-react) stating that there should be three layers of validation:
- On the DB level,
- On the Backend level
- On the frontend level

Reply:
About the default language. I do not think it should be a requirement to always have a a default language. But We should make sure we don't have more than one. But definitely we should throw an error when we are trying to retrieve the default language but it doesn't exist in the DB.